src/storage/sqlite.rs

```diff
@@ -249,6 +252,35 @@ impl SqliteStorage {
         }
         Ok(objid)
     }
+
+    fn num_to_val(ulong: CK_ULONG) -> error::Result<Value> {
+        /* CK_UNAVAILABLE_INFORMATION need to be special cased */
+        /* for storage compatibility CK_ULONGs can only be stored as u32
+         * values and PKCS#11 spec pay attentions to never allocate numbers
+         * bigger than what can be stored in a u32. However the value of
+         * CK_UNAVAILABLE_INFORMATION is defined as CK_ULONG::MAX which is
+         * a larger number than what we can store in a u32.
+         * Sqlite however can store bugger than u32 numbers, so we store
```

Comment:
```suggestion
         * Sqlite however can store bigger than u32 numbers, so we store
```

Reply:
Ah nice you pointed this out, as this explanation is actually outdated.
This was the explanation for the first attempt, I since changed the code to store -1 instead (still with the same exception approach for otherwise handling the case), which is more fitting.